utilities/installer/install_puppet.rb

```diff
@@ -130,52 +143,45 @@
 end
 
 GSUTILITY = '/isan/bin/guestshell'
-TMP_PUPPET_LOCATION = '/bootflash/tmp_puppet'
 PUPPET_PATH = '/opt/puppetlabs/puppet'
 
 # -------------------------#
 # Bootstrap Helper Methods #
 # -------------------------#
 
-# Generate commands to prepend to native or
-# guestshell targets
-#
-# @param host [String]
-# @return [String] Network namespace command
-# @example on agent, "#{pp(agent)} puppet agent -t"
-def pp(host)
+# Temporary file storage location on the target
+def tmp_location(host)
   case host['platform']
-  when /cisco/
-    if host['vrf'].nil?
-      command = 'sudo'
-    else
-      if target_guestshell?(host)
-        command = "sudo chvrf #{host['vrf']}"
-      else
-        command = "sudo ip netns exec #{host['vrf']}"
-      end
-    end
-    command.prepend "sudo #{GSUTILITY} " if target_guestshell?(host)
-    command.prepend "http_proxy=#{options['http_proxy']} " if
-                     options['http_proxy']
-    command.prepend "https_proxy=#{options['https_proxy']} " if
-                     options['https_proxy']
-    command.prepend "PATH=$PATH:#{PUPPET_PATH}/bin/:#{PUPPET_PATH}/lib/ " unless
-                     target_guestshell?(host)
-  when /other_platform/
-    command = 'do something different'
-  else
-    command = ''
+  when /cisco_nexus/
+    '/bootflash/tmp_puppet'
+  when /cisco_ios_xr/
+    '/disk0:/tmp_puppet'
   end
-  command
 end
 
+def create_tmp_location(agent)
+  on agent, "mkdir -p #{tmp_location(agent)}"
+  on agent, "chmod a+rw #{tmp_location(agent)}"
+end
+
+def destroy_tmp_location(agent)
+  on agent, "rm -rf #{tmp_location(agent)}"
+end
+
+$env = {}
+$env['http_proxy'] = options['http_proxy'] if options['http_proxy']
+$env['https_proxy'] = options['https_proxy'] if options['https_proxy']
+$env['PATH'] = "$PATH:#{PUPPET_PATH}/bin/:#{PUPPET_PATH}/lib/"
+
+# TODO: if installing to guestshell:
+# command.prepend "#{GSUTILITY} sudo chvrf ${host['vrf']}"
```

Comment:
Just want to call attention to this - now that Beaker is doing the 'ip netns exec' command for us, if guestshell still isn't supporting 'ip netns exec' but instead requires 'chvrf', beaker probably won't work against guestshell. Someone probably needs to investigate this. :-)